eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -169,31 +184,33 @@ public QueryResponse getAnnotations(@PathVariable("variantCoreString") String va
         }
         MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species + "_" + assembly));
 
-        List<VariantWithSamplesAndAnnotation> variantEntities;
+        Optional<VariantWithSamplesAndAnnotation> variantEntity;
         try {
-            variantEntities = getVariantEntitiesByParams(variantCoreString, annotationVepVersion,
+            variantEntity = getVariantByCoordinatesAndAnnotationVersion(variantCoreString, annotationVepVersion,
                     annotationVepCacheVersion);
-            if (variantEntities.size() == 0) {
-                response.setStatus(HttpServletResponse.SC_NOT_FOUND);
-                return setQueryResponse(buildQueryResult(variantEntities, 0));
-            }
         } catch (AnnotationMetadataNotFoundException ex) {
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
             return setQueryResponse(ex.getMessage());
+        } catch (IllegalArgumentException ex) {
```

Comment:
Again, multiple exception types that are handled identically can be grouped in a single `catch` block.

Reply:
Fixed :)